task/sast-find-unicode-control/0.1/sast-find-unicode-control.yaml

```diff
@@ -0,0 +1,232 @@
+            fi
+        
+            # translate the output format
+            sed -i raw_sast_find_unicode_control_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                show_error
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format\n" >&2
+                SCAN_RESULT="raw_sast_find_unicode_control_check_out.txt"
+            fi
+        }
+        
+        # Filter known false positive
+        filter_kfp()
+        {
+            # We check if the KFP_GIT_URL variable is set to apply the filters or not
+            if [[ -z "${KFP_GIT_URL}" ]]; then
+                echo "KFP_GIT_URL variable not defined. False positives won't be filtered" >&2
+                mv sast_snyk_check_out_all_findings.json filtered_sast_snyk_check_out.json
+                return 0
+            else
+                echo "Filtering false positives in results files using csfilter-kfp..." >&2
+                cmd=（
+                    csfilter-kfp
+                    --verbose 
+                    --kfp-git-url="${KFP_GIT_URL}"
+                ）
+
+                if [[ -z "${PROJECT_NVR}" ]]; then
+                    cmd+=(--project-nvr="${PROJECT_NVR}")
+                if
+
+                if [[ -z "${RECORD_EXCLUDED}" ]]; then
+                    cmd+=(--record-excluded="${RECORD_EXCLUDED}")
+                if
+
+                "${cmd[@]}" raw_sast_find_unicode_control_check_out.txt > filtered_sast_find_unicode_control_check_out.json
+                status=$?
+                if [ "$status" -ne 0 ]; then
+                    echo "Error: failed to filter known false positives" >&2
+                    show_error
+                    return 1
+                else
+                    echo "Message: Succeed to filter known false positives" >&2
+                    SCAN_RESULT="filtered_sast_find_unicode_control_check_out.json"
+                fi
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --event='error|warning'
+                --remove-duplicates
+                --embed-context=3
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as Snyk only provides an URI to embed the context
+            set -x
+            csgrep --mode=json --prepend-path-prefix="${SOURCE_CODE_DIR}"/ "${SCAN_RESULT}" | csgrep "${CSGERP_OPTS[@]}" | csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/  > processed_sast_find_unicode_control_check_out.json
+            set +x
```

Comment:
I would not disable the XTRACE mode here.  If something fails in the subsequent commands, recording the commands that have been executed is pretty useful for debugging.

Reply:
yes, I have updated the code, set -x always